FirebaseInstallations/Source/Library/InstallationsStore/FIRInstallationsStoredItem.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+
+@class FIRInstallationsStoredAuthToken;
+
+NS_ASSUME_NONNULL_BEGIN
+
+typedef NS_ENUM(NSInteger, FIRInstallationsStatus) {
+  // Represents either an initial status when a FIRInstallationsItem instance was created but not
+  // stored to Keychain or an undefined status (e.g. when the status failed to deserialize).
+  FIRInstallationStatusUnknown,
+  // The Firebase Installation has not yet been registered with FIS.
+  FIRInstallationStatusUnregistered,
+  // #CreateInstallation request to FIS server-API is in progress.
+  FIRInstallationStatusRegistrationInProgress,
+  // The Firebase Installation has successfully been registered with FIS.
+  FIRInstallationStatusRegistered,
+};
+
+@interface FIRInstallationsStoredItem : NSObject <NSSecureCoding>
```

Comment:
what is the difference between FIRInstallationsStoredItem and FIRInstallationsItem? Would it be possible to merge these two classes. I think part of the pain point of understanding IID is that there are many layers of classes making the code hard to read. I wonder if we can start from simple and expand.

Reply:
@chliangGoogle  We've had discussion around this in the [design doc](https://docs.google.com/a/google.com/document/d/1max117focXmar6mRxs0MDRD0H9OGlBf80fmxI1oJLNQ/edit?disco=AAAADDdu2Qs). The `*StoredItems` classes are the objects that will be stored into Keychain as is. So, basically, `FIRInstallationsStoredItem` is kind of a storage scheme. In contrast `FIRInstallationsItem` is the model that is used by the logic layer and contains some logic as well. 
Another point - the `FIRInstallationsStoredItem` needs to be stable to avoid the storage migrations, from the other side the logic layer may require updates, convenience methods, properties, etc. Having separate classes for the logic and for the store allows to satisfy both. 

Taking this into account, I would prefer to have separate `FIRInstallationsStoredItem` and `FIRInstallationsItem` unless there is a significant downside of it. We chat offline on this topic if you want.

Reply:
But there's no API level of preventing unexpected extra properties added to FIRInstallationsStoredItem, I guess we can comment in the class to add notice to be cautious to add any new items to this class.  I'm not convinced for scheme to duplicate classes who have almost identical properties and usage. You can always pick and choose the properties to store afaik.

But I don't feel strong about it, I do like the clarification on the design doc per class if you can add them in the code too if you prefer to keep both.

Reply:
@chliangGoogle I added the docs to the class with some explanation and warnings on the stored classes modifications. We still may revisit this questions later on before the release. If we decide to get rid of the separate classes, it will involve mostly renaming and some minor modifications in the storage.